pymc3/distributions/multivariate.py

```diff
@@ -93,35 +61,64 @@ class MvNormal(Continuous):
 
     """
 
-    def __init__(self, mu, cov=None, tau=None, gpu_compat=False, *args, **kwargs):
+    def __init__(self, mu, cov=None, tau=None, chol=None, gpu_compat=False,
+                 *args, **kwargs):
         super(MvNormal, self).__init__(*args, **kwargs)
+        if len([i for i in [tau, cov, chol] if i is not None]) != 1:
+            raise ValueError('Incompatible parameterization. Specify exactly '
+                             'one of tau, cov, or chol to specify '
+                             'distribution.')
         self.mean = self.median = self.mode = self.mu = mu = tt.as_tensor_variable(mu)
-        tau, cov = get_tau_cov(mu, tau=tau, cov=cov)
-        self.tau = tt.as_tensor_variable(tau)
-        self.cov = tt.as_tensor_variable(cov)
+        self.solve = tt.slinalg.Solve(A_structure="lower_triangular", lower=True)
+
+        # Use the Cholesky logp unless tau is given
+        self.use_chol_logp = tau is None
+        if cov is not None:
+            self.chol = tt.slinalg.cholesky(tt.as_tensor_variable(cov))
+        elif tau is not None:
+            self.tau = tt.as_tensor_variable(tau)
+            self.chol = tt.slinalg.cholesky(tt.nlinalg.matrix_inverse(tau))
```

Comment:
I'd rather this wasn't here. Maybe just set `self.chol` to None? I can't think of a legitimate reason to even use this, as the inverse is not stable, and I don't think theano will optimize it away.